src/gadgets/nonnative/bignat.rs

```diff
@@ -444,7 +444,11 @@ impl<Scalar: PrimeField> BigNat<Scalar> {
     let carry_bits = (((max_word.to_f64().unwrap() * 2.0).log2() - self.params.limb_width as f64)
       .ceil()
       + 0.1) as usize;
-    let limbs_per_group = (Scalar::CAPACITY as usize - carry_bits) / self.params.limb_width;
+    let limbs_per_group = max(
+      (Scalar::CAPACITY as usize - carry_bits) / self.params.limb_width,
+      1,
+    );
+
```

Comment:
I am not sure of the exact arithmetic, but would 
```rust
let limbs_per_group = (Scalar::CAPACITY as usize - carry_bits + (self.params.limb_width-1)) / self.params.limb_width
```
may be closer to the intended behavior.

Reply:
My intention with the change was to ensure that the number of limbs per group is at least 1 without modifying the original behavior of the circuit. This came up in the case with BN width of 128 and limb number 2 where the original calculation results in `limbs_per_group = 0`, leading to a divide by zero panic when calculating `n_limbs` in `group_limbs(...)`. 

With the change you propose the `limbs_per_group` matches for the 128 x 2 configuration, but also has the effect of changing the value of `limbs_per_group` in other cases as well. In particular in the 64 x 4 configuration this leads to 3 `limbs_per_group` compared to the original 2. Interestingly this leads to a slightly smaller circuit, but it also causes tons of panics in tests checking IVC proof/verification, so I think it causes some issues with the original logic.

Reply:
Hmm, I think I didn't correctly understand the intention of this value. My suggestions was essentially performing a `ceil` of the `(Scalar::CAPACITY as usize - carry_bits) / self.params.limb_width` value in a safe way, but I don't think that was actually intended here .